### ISSUE TITLE
Is there a way to upgrade from 18.10 to 19.04 now?

### ISSUE THREAD

**Issue by hjchai** [pop-os/pop#669]
**Distribution (run `cat /etc/os-release`):**
 
NAME="Pop!_OS"
VERSION="18.10"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Pop!_OS 18.10"
VERSION_ID="18.10"
HOME_URL="https://system76.com/pop"
SUPPORT_URL="http://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=cosmic
UBUNTU_CODENAME=cosmic

 
 
**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):**

**Issue/Bug Description:**

Can not upgrade from 18.10 to 19.04 since 18.10 is EOL. 

**Steps to reproduce (if you know):**

`sudo apt full-upgrade`

**Expected behavior:**

...
Err:1 http://apt.pop-os.org/proprietary cosmic/main amd64 atom amd64 1.38.2
  404  Not Found
...

**Other Notes:**

Basically, I did similar things as this post states: https://www.reddit.com/r/pop_os/comments/d8g5tj/having_trouble_upgrading_pop_os_from_1810_i_know/

Seems like there is no way to do the upgrade now except for a clean install? Does anyone have similar experience? Thanks in advance.

**Comment by msbrowning** [comment]
@hjchai I just had to update from 18.10 to 19.04 to update to 19.10 this morning. It took me a minute to find the documentation but if you look in the 'Upgrading older releases' section at https://support.system76.com/articles/upgrade-pop/ you'll find the instructions. 

Basically it was sudo apt update, sudo apt full-upgrade, do-release-upgrade and then I was at 19.04.

Hope this helps!

**Comment by hjchai** [comment]
@msbrowning Hi, thanks for the reply. Are you certain you were at 18.10 previously? popos team removed repository server for 18.10. For that reason, when I do sudo apt update, I got 'E: The repository 'http://apt.pop-os.org/proprietary cosmic Release' no longer has a Release file.'

With this error, the following steps just don't work properly. Could it possible that you were at 18.04 before? Since 18.04 is a LTS, Popos still maintain updates to the repository server. And you can upgrade to whatever version of popos from there. Just my wild guess. Please let me know.

**Comment by msbrowning** [comment]
I just went through my /var/log/dpkg.log and I was indeed on 18.10 and from 19.04 on the 21st. It took several tries but I was able to update it before updating to 19.10.

**Comment by hjchai** [comment]
@msbrowning Hi Mark, that's really exciting to know. Do you mind sharing what specific "tries" you did to make it work? Any special tricks? That will be really helpful to me. So far, I have no luck to by-pass the error I mentioned. I have been stuck in 18.10 for weeks. Some of applications can not updates in an EOL OS, that's upsetting...

**Comment by mmstick** [comment]
If you're still on 18.10, you should reinstall with the 19.04 ISO.
